specification/developerhub/resource-manager/Microsoft.DevHub/preview/2024-05-01-preview/iac.json

```diff
@@ -655,6 +655,10 @@
           {
             "value": "HCIARCVM",
             "description": "The template use quick start template of HCI and ArcVM"
+          },
+          {
+            "value": "ALL",
+            "description": "The template use quick start template of All supported products"
```

Comment:
Does "ALL" include every future value in the enum or not? If you introduce new values in future API versions, do resources created using previous API versions with "ALL" automatically compatible with the new values?

Reply:
both are yes! Our service is used to help customer prepare their IaC(Infrastructure as Code) repo. When a customer puts a profile or Post Scale/export will end up with a PR generated for their repo. When putting a new template, choosing all enum means our service generates PR for customers with all supported products. The difference between the enum types will be in the content of the generated PR. So ALL is aimed to be compatible with new values. Thank you for the review:)